graypy/handler.py

```diff
@@ -62,7 +64,49 @@ def makePickle(self, record):
         message_dict = make_message_dict(
             record, self.debugging_fields, self.extra_fields, self.fqdn,
             self.localname, self.level_names, self.facility)
-        return zlib.compress(message_to_pickle(message_dict))
+        packed = message_to_pickle(message_dict)
+        frame = zlib.compress(packed) if self.compress else packed
+        return frame
+
+
+class GELFTcpHandler(SocketHandler):
+    """Graylog Extended Log Format handler over TCP
+
+    :param host: The host of the graylog server.
+    :param port: The port of the graylog server (default 12201).
+    :param debugging_fields: Send debug fields if true (the default).
+    :param extra_fields: Send extra fields on the log record to graylog
+        if true (the default).
+    :param fqdn: Use fully qualified domain name of localhost as source
+        host (socket.getfqdn()).
+    :param localname: Use specified hostname as source host.
+    :param facility: Replace facility with specified value. If specified,
+        record.name will be passed as `logger` parameter.
+    :param level_names: Allows the use of string error level names instead
+        of numerical values. Defaults to False
+    :param compress: Allows to use message compression. Defaults to False
+    """
+
+    def __init__(self, host, port=12201,
+                 debugging_fields=True, extra_fields=True, fqdn=False,
+                 localname=None, facility=None, level_names=False, compress=False):
+        self.debugging_fields = debugging_fields
+        self.extra_fields = extra_fields
+        self.fqdn = fqdn
+        self.localname = localname
+        self.facility = facility
+        self.level_names = level_names
+        self.compress = compress
+        SocketHandler.__init__(self, host, port)
+
+    def makePickle(self, record):
```

Comment:
Can we DRY this? Maybe make a class and then make it inherited both by `GELFHandler` and  `GELFTcpHandler` .